### PR TITLE
Update ShipmentMailer to use Cartons

### PR DESCRIPTION
**Pull request by philbirt**
* Move ShippingManifest into a module that can be shared by shipments and cartons

### CODE REVIEW

File: core/app/mailers/spree/shipment_mailer.rb [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
   class ShipmentMailer < BaseMailer
-    def shipped_email(shipment, resend = false)
-      @shipment = shipment.respond_to?(:id) ? shipment : Spree::Shipment.find(shipment)
+    def shipped_email(carton, resend = false)
+      @carton = carton.respond_to?(:id) ? carton : Spree::Carton.find(carton)
       subject = (resend ? "[#{Spree.t(:resend).upcase}] " : '')
-      subject += "#{Spree::Config[:site_name]} #{Spree.t('shipment_mailer.shipped_email.subject')} ##{@shipment.order.number}"
-      mail(to: @shipment.order.email, from: from_address, subject: subject)
+      subject += "#{Spree::Config[:site_name]} #{Spree.t('shipment_mailer.shipped_email.subject')} ##{@carton.order_numbers}"
+      mail(to: @carton.orders.first.email, from: from_address, subject: subject)
```
Comment (jordan-brough) [inline core/app/mailers/spree/shipment_mailer.rb]: how about `mail(to: @carton.order_emails, ...`? (where #order_emails returns a unique list of email addrs)  That seems like it fits the many-to-many relationship better right?

File: core/app/models/spree/carton.rb [hunk 2]
```diff
@@ -25,4 +27,8 @@ def to_param
   def tracking_url
     @tracking_url ||= shipping_method.build_tracking_url(tracking)
   end
+
+  def order_numbers
+    orders.map(&:number).join(',')
```
Comment (jordan-brough) [inline core/app/models/spree/carton.rb]: i'd vote to put the `join` in the view layer and let this return an array

File: core/app/models/spree/order_shipping.rb [hunk 3]
```diff
@@ -58,6 +58,7 @@ def ship(inventory_units:, stock_location:, address:, shipping_method:,
       )
     end
 
+
```
Comment (jordan-brough) [inline core/app/models/spree/order_shipping.rb]: intentional?

File: core/app/mailers/spree/shipment_mailer.rb (hunk 1 above)
Comment (philbirt) [inline core/app/mailers/spree/shipment_mailer.rb]: I like that idea.

File: core/app/models/spree/carton.rb (hunk 2 above)
Comment (philbirt) [inline core/app/models/spree/carton.rb]: Cool, will do

File: core/app/models/spree/order_shipping.rb (hunk 3 above)
Comment (philbirt) [inline core/app/models/spree/order_shipping.rb]: Will remove

**Comment by jordan-brough** [comment]
:+1:

**Comment by jordan-brough** [comment]
got a :+1: from @jhawthorn via the solidus chatroom

**Pull request merged**